partner_merge_group/wizard/partner_merge_views.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, _
+
+
+class MergePartnerAutomatic(models.TransientModel):
+    _inherit = 'base.partner.merge.automatic.wizard'
+
+    def _merge(self, partner_ids, dst_partner=None):
+        super(MergePartnerAutomatic, self).sudo(self.user.id).with_context(
+            merge_with_original_user_id=self.env.uid
+        )._merge(self, partner_ids, dst_partner)
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.multi
+    def message_post(self, body, subject, message_type, subtype,
+                     parent_id, attachments, content_subtype, **kwargs):
+        if self.env.context.get('merge_with_original_user_id'):
+            body += _("\n\nMerge processed by {}.".format(self.env.user.name))
```

Comment:
The point of passing `merge_with_original_user_id` is to browse the id and display the corresponding name. Being called with `sudo`, you will always have `admin` if you use `self.env.user.name`.